src/main/groovy/liquibase/harness/compatibility/advanced/AdvancedTest.groovy

```diff
@@ -0,0 +1,188 @@
+        boolean shouldRunChangeSet
+
+        and: "check database under test is online"
+        def connection = testInput.database.getConnection()
+        shouldRunChangeSet = connection instanceof JdbcConnection
+        assert shouldRunChangeSet: "Database ${testInput.databaseName} ${testInput.version} is offline!"
+
+        and: "ignore testcase if it's invalid for this combination of db type and/or version"
+        shouldRunChangeSet = !expectedSql?.toLowerCase()?.contains("invalid test")
+        Assume.assumeTrue(expectedSql, shouldRunChangeSet)
+
+        and: "configuring test data for generateChangelog command for all files format"
+        def shortDbName = getShortDatabaseName(testInput.databaseName)
+        def generateChangelogMap = configureChangelogMap(testInput.generateCLResourcesPath, shortDbName)
+        def diffChangelogMap = configureChangelogMap(testInput.diffCLResourcesPath, shortDbName)
+
+        when: "get sql generated for the change set"
+        def generatedSql = parseQuery(executeCommandScope("updateSql", argsMapPrimary).toString())
+
+        then: "verify expected sql matches generated sql"
+        validateSql(generatedSql, expectedSql)
+
+        when: "apply changeSet to DB"
+        executeCommandScope("update", argsMapPrimary)
+
+        then: "get DB snapshot, check if actual snapshot matches expected snapshot"
+        def generatedSnapshot = executeCommandScope("snapshot", argsMapPrimary).toString()
+        snapshotMatchesSpecifiedStructure(testInput.expectedSnapshot, generatedSnapshot)
+
+        for (Map.Entry<String, String> entry : generateChangelogMap.entrySet()) {
+
+            when: "execute generateChangelog command using different changelog formats"
+            argsMapPrimary.put("changelogFile", resourcesDirFullPath + entry.value)
+            executeCommandScope("generateChangelog", argsMapPrimary)
+
+            then: "check if a changelog was actually generated and validate it's content"
+            String generatedChangelog = parseQuery(readFile((String) argsMapPrimary.get("changelogFile")))
+            validateChangelog(entry.key, generatedChangelog, testInput.verificationGenCLSql, testInput.change, null)
+
+            when: "execute updateSql command on generated changelogs"
+            argsMapPrimary.put("changelogFile", resourcesDirPath + entry.value)
+            generatedSql = parseQuery(executeCommandScope("updateSql", argsMapPrimary).toString())
+            generatedSql = removeSchemaNames(generatedSql, testInput.database, testInput.primaryDbSchemaName)
+
+
+            then: "execute updateSql command on generated changelogs"
+            expectedSql = getChangelogValidationSql("generateChangelog", testInput.change, testInput.databaseName, testInput.version)
+            validateSql(generatedSql, expectedSql)
+        }
+
+        when: "execute diff command"
+        String generatedDiff = cleanDiff(executeCommandScope("diff", argsMapSecondary).toString())
```

Comment:
Please rename the method 'cleanDiff' to something like 'getCleanDiff', or another name that will better reflect the functionality of this method.